x-pack/platform/plugins/shared/streams/server/lib/streams/helpers/normalize_geo_points.ts

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isPlainObject } from 'lodash';
+
+export interface GeoPointValue {
+  lat: unknown;
+  lon: unknown;
+}
+
+export function normalizeGeoPointsInObject(
+  obj: Record<string, unknown>,
+  geoPointFields: Set<string>
+): Record<string, unknown> {
+  function normalizeRecursive(current: unknown, path = ''): unknown {
```

Comment:
It seems like in the places this helper is used, we eventually flatten the object, so we don't need to make this recursive and keep things simpler. Could you adjust?